src/ESLintLinter.php

```diff
@@ -45,7 +52,40 @@ public function getLinterConfigurationName() {
   }
 
   public function getDefaultBinary() {
-    return 'eslint';
+    if ($this->cwd) {
+      list($err, $stdout, $stderr) = exec_manual('yarn -s --cwd %s bin', $this->realCwd);
+      if ($stdout) {
+        $localBinaryPath = strtok($stdout, "\n") . "/eslint";
+      }
+    } else {
+      $localBinaryPath = './node_modules/.bin/eslint';
+    }
+
+    if (file_exists($localBinaryPath)) {
+      return $localBinaryPath;
+    }
+
+    if (!$this->cwd) {
+      // Fallback on global install
+      // Note: Arcanist checks binary existence internally
+      list($err, $stdout, $stderr) = exec_manual('which eslint');
```

Comment:
Consider using `Filesystem::binaryExists('eslint')` instead. That might simplify some of the code here.

From `libphutil/src/filesystem/Filesystem.php`:

```php
  /**
   * Determine if an executable binary (like `git` or `svn`) exists within
   * the configured `$PATH`.
   *
   * @param   string  Binary name, like `'git'` or `'svn'`.
   * @return  bool    True if the binary exists and is executable.
   * @task    exec
   */
  public static function binaryExists($binary) {
    return self::resolveBinary($binary) !== null;
  }


  /**
   * Locates the full path that an executable binary (like `git` or `svn`) is at
   * the configured `$PATH`.
   *
   * @param   string  Binary name, like `'git'` or `'svn'`.
   * @return  string  The full binary path if it is present, or null.
   * @task    exec
   */
  public static function resolveBinary($binary) {
    if (phutil_is_windows()) {
      list($err, $stdout) = exec_manual('where %s', $binary);
      $stdout = phutil_split_lines($stdout);

      // If `where %s` could not find anything, check for relative binary
      if ($err) {
        $path = self::resolvePath($binary);
        if (self::pathExists($path)) {
          return $path;
        }
        return null;
      }
      $stdout = head($stdout);
    } else {
      list($err, $stdout) = exec_manual('which %s', $binary);
    }

    return $err === 0 ? trim($stdout) : null;
  }
```